04/app.js

```diff
@@ -1,40 +1,51 @@
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener("DOMContentLoaded", init);
 
-function init() {
-    const boxElement = document.querySelector('.box');
-    setBoxShadow(boxElement, '#000000');
+function init(e) {
+  const boxElement = document.querySelector(".box");
+  setBoxShadow(boxElement, e.target.value);
 }
 
 function setBoxShadow(element, colorInHex, opacity = 1) {
-    const colorInRGBA = `rgba(
-        ${getChannelColor(colorInHex, 'red')}, 
-        ${getChannelColor(colorInHex, 'green')}, 
-        ${getChannelColor(colorInHex, 'blue')}, 
+  const colorInRGBA = `rgba(
+        ${getChannelColor(colorInHex, "red")}, 
+        ${getChannelColor(colorInHex, "green")}, 
+        ${getChannelColor(colorInHex, "blue")}, 
         ${opacity}
     )`;
 
-    element.style.boxShadow = `0 0 5px 5px ${colorInRGBA}`;
+  element.style.boxShadow = `0 0 5px 5px ${colorInRGBA}`;
+  console.log(element);
+  console.log(colorInRGBA);
+  console.log(opacity);
 }
 
-
 function getChannelColor(colorInHex, channelName) {
-    let start;
-    switch(channelName) {
-        case 'red':
-            start = 1;
-            break;
-        case 'green':
-            start = 3;
-            break;
-        case 'blue':
-            start = 5;
-            break;
-    }
-
-    const channelColorHex = colorInHex.substr(start, 2);
-    const channelColorDec = parseInt(channelColorHex, 16);
-
-    return channelColorDec; 
+  let start;
+  switch (channelName) {
+    case "red":
+      start = 1;
+      break;
+    case "green":
+      start = 3;
+      break;
+    case "blue":
+      start = 5;
+      break;
+  }
+
+  const channelColorHex = colorInHex.substr(start, 2);
+  const channelColorDec = parseInt(channelColorHex, 16);
+  console.log(channelColorDec);
+
+  return channelColorDec;
 }
 
+const colorInput = document.querySelector("input");
+const rangeInput = colorInput.nextElementSibling;
+
+//rangeInput.addEventListener("mousemove", init);
+rangeInput.addEventListener("change", init);
+colorInput.addEventListener("change", init);
```

Comment:
Napisz 2 różne funkcje, możesz to zrobić bezpośrednio i tam wywołac funkcję z odpowiednimi argumentami tj.
```
rangeInput.addEventListener("change", function {
    // ...
    setBoxShadow(boxElement, color, opacity);
});
```

PS. Jeśli ta podpowiedź nie wystarczy to napisz na Slacku do mnie ;)